File: app/api/routes/sessions.py
"""Sessions REST API — CRUD operations for learning sessions."""

import uuid
from datetime import datetime, timezone

from fastapi import APIRouter

from app.models.session import SessionCreate, SessionResponse

router = APIRouter()

# In-memory fallback (replaced by DB repositories in Phase 4 with real PostgreSQL)
_sessions: dict[str, dict] = {}


@router.get("/", response_model=dict)
async def list_sessions():
    return {
        "sessions": [
            {
                "id": s["id"],
                "thread_id": s["thread_id"],
                "learning_goal": s["learning_goal"],
                "progress": s["progress"],
                "status": s["status"],
                "created_at": s["created_at"],
            }
            for s in _sessions.values()
        ]
    }


@router.post("/", response_model=dict)
async def create_session(body: SessionCreate):
    sid = str(uuid.uuid4())
    thread_id = str(uuid.uuid4())
    now = datetime.now(timezone.utc)

    session = {
        "id": sid,
        "thread_id": thread_id,
        "learning_goal": body.learning_goal,
        "progress": 0.0,
        "status": "active",
        "created_at": now.isoformat(),
    }
    _sessions[sid] = session
    return session


@router.delete("/{session_id}")
async def delete_session(session_id: str):
    _sessions.pop(session_id, None)
    return {"id": session_id, "status": "deleted"}

File: frontend/src/api/sessions.ts
import type { SessionResponse, SessionCreate } from './types'

const BASE = '/sessions/'

export async function listSessions(): Promise<SessionResponse[]> {
  const res = await fetch(BASE)
  if (!res.ok) throw new Error(`获取会话列表失败: ${res.status}`)
  const data = await res.json()
  return data.sessions ?? []
}

export async function createSession(
  input: SessionCreate,
): Promise<SessionResponse> {
  const res = await fetch(BASE, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(input),
  })
  if (!res.ok) throw new Error(`创建会话失败: ${res.status}`)
  return res.json()
}

export async function deleteSession(sessionId: string): Promise<void> {
  const res = await fetch(`${BASE}/${sessionId}`, { method: 'DELETE' })
  if (!res.ok) throw new Error(`删除会话失败: ${res.status}`)
}
